### ISSUE TITLE
IndexOutOfBoundsException in FunctionInvoker

### ISSUE THREAD

**Issue by wenqiglantz** [spring-cloud/spring-cloud-function#929]
**Describe the bug**
My app is built on Spring Boot 2.7.4, Spring Cloud Function 3.2.7, Feign Reactive 3.2.6, and Java 11.  I have a function which calls a third party vendor's DELETE API call, returning `Mono<Void>`, as that API doesn't have a response body returning.  I ran into the following error when executing this function.

```
Index 0 out of bounds for length 0: java.lang.IndexOutOfBoundsException java.lang.IndexOutOfBoundsException: 
Index 0 out of bounds for length 0  at java.base/jdk.internal.util.Preconditions.outOfBounds(Unknown Source)  at 
java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Unknown Source)  at 
java.base/jdk.internal.util.Preconditions.checkIndex(Unknown Source)  at 
java.base/java.util.Objects.checkIndex(Unknown Source)  at 
java.base/java.util.ArrayList.get(Unknown Source)  at 
org.springframework.cloud.function.adapter.aws.FunctionInvoker.buildResult(FunctionInvoker.java:106)  at 
org.springframework.cloud.function.adapter.aws.FunctionInvoker.handleRequest(FunctionInvoker.java:86)  
```

Looking at the source code for `FunctionInvoker`, see below the relevant block.  The `else` block threw the actual error.  In my use case, since my response from the DELETE API call returned nothing, the result List would still be an empty list, it doesn't meet condition `if (result.size() > 1)`, falling into the `else` block, only to throw the above exception.  I believe the fix needs to be at the `if` statement to revise it into something like `if (result.size() != 1)`.  Let me know if you can provide a better fix.

```java
if (output instanceof Publisher<?>) {
	List<Object> result = new ArrayList<>();
	for (Object value : Flux.from((Publisher<?>) output).toIterable()) {
		if (logger.isInfoEnabled()) {
			logger.info("Response value: " + value);
		}
		result.add(value);
	}
	if (result.size() > 1) {
		output = result;
	}
	else {
		output = result.get(0);
	}
....
}
```

**Comment by olegz** [comment]
I see the issue and thank you for reporting it. And yes you are correct about "not complete" IF statement. 
So, it effectively should behave as Consumer which we already support

**Comment by olegz** [comment]
The issue has been fixed and back-ported and we'll have another 3.2.x release by the end of the month

**Comment by wenqiglantz** [comment]
Thanks @olegz!  Appreciate it!
